.github/workflows/go-test.yml

```
@@ -0,0 +1,51 @@
+name: Go tests
+on:
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - windows-latest
+          - ubuntu-latest
+        go-version:
+          - 1.19.x
+          - 1.22.x
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Configure git in Windows.
+        run: |
+          git config --global core.autocrlf input
+        if: ${{ matrix.os == 'windows-latest' }}
+
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Setup Go ${{ matrix.go-version }}.
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Static checks with Go vet.
+        run: go vet ./...
+
+      - name: Static checks with Staticcheck.
+        run: go run honnef.co/go/tools/cmd/staticcheck ./...
+
+      - name: Test with the Go CLI.
+        run: go test -v ./...
+
+      - name: Check pending updates.
+        run: |
+          go run golang.org/x/tools/cmd/goimports -w .
```

I think there is no local import yet, but it could be added here the local flag in case an internal or local package is added in the future:

```suggestion
          go run golang.org/x/tools/cmd/goimports -local github.com/elastic/go-resource/ -w .
```